aerogear/mobile-security-service
# feat(init): Create /init endpoint for POST (AEROGEAR-8360)

## Motivation

https://issues.jboss.org/browse/AEROGEAR-8360

## What

Added init endpoint.

## Why

Required to return information to the app.

## Verification Steps

Send a POST request to `/api/init` with the following data:

```json
{
	"deviceId": "d1895cc1-28d7-4283-932d-8bcab9e4a469",
	"deviceVersion": "3.4",
	"version": "3.10",
	"deviceType": "Android",
	"appId": "com.aerogear.testapp"
}
```

You should receive the init data which looks something like this:

```json
{
    "id": "4ca6376c-61cc-4c5f-8a47-f17dd8b9e624",
    "version": "3.10",
    "appId": "com.aerogear.testapp",
    "disabled": true,
    "disabledMessage": "DISABLED!!"
}
```

Change the version in the original POST request. After POSTing again you should still get back data. A new version was created in the database.

Check the version table for that version. Is the `num_of_app_launches` counter increasing every time you make a request?

## Checklist:


## Progress


## Additional Notes

This task is not finished. The repository unit tests have not been completed. Please review what is there.
